specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,264 @@
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+@useAuth(
+  OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize",
+
+      // add description to below: "impersonate your user account"
+      scopes: ["https://eventhubs.azure.net/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Azure Schema Registry",
+})
+
+// Supported operations.
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+//  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{version}?api-version={apiVersion}
+//  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/:get-id?api-version={apiVersion}
+//  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+@server(
+  "{fullyQualifiedNamespace}",
+  "The Schema Registry service endpoint.",
+  {
+    @doc("The Schema Registry service endpoint, for example 'https://my-namespace.servicebus.windows.net'.")
+    fullyQualifiedNamespace: url,
+  }
+)
+@versioned(ServiceApiVersions)
+@doc("SchemaRegistryClient is a client for registering and retrieving schemas from the Azure Schema Registry service.")
+namespace SchemaRegistry;
+
+@doc("Represents the Schema Registry API version to use for requests.")
+enum ServiceApiVersions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2023-07-01 Version. This is the default version.")
+  V2023_07_01: "2023-07-01",
```

Comment:
The older service versions aren't in here anymore.  Is there a transform you apply to this?
Example:
* [SchemaRegistryVersion](https://learn.microsoft.com/en-us/java/api/com.azure.data.schemaregistry.schemaregistryversion?view=azure-java-stable#fields)
* [SchemaRegistryClientOptions.ServiceVersions](https://learn.microsoft.com/en-us/dotnet/api/azure.data.schemaregistry.schemaregistryclientoptions.serviceversion?view=azure-dotnet#fields)

Reply:
OOB discussion: Adding back in + renaming to ServiceVersions